### PR TITLE
Add FxCop analyzers to build

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

**Comment by bartonjs** [comment]
Have the suppressions already been added to the projects that needed them?  It's not good when buildtools upgrades get blocked.

**Comment by morganbr** [comment]
The list of checks looks good. I don't know enough about the build system to review that part.

**Comment by alexperovich** [comment]
The suppressions were added in dotnet/corefx#13014

**Comment by weshaggard** [comment]
cc @AlexGhiondea

**Comment by mellinoe** [comment]
Is there much of an impact on build time with this?

File: src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets [hunk 1]
```diff
@@ -21,7 +21,18 @@
...
+    <!-- %24 = $ -->
+    <EnableFxCopAnalyzers Condition="$([System.Text.RegularExpressions.Regex]::IsMatch($(MSBuildProjectDirectory), 'src%24'))">true</EnableFxCopAnalyzers>
+    <EnableFxCopAnalyzers Condition="'$(EnableFxCopAnalyzers)' != 'true'">false</EnableFxCopAnalyzers>
+  </PropertyGroup>
+
+  <ItemGroup Condition="$(EnableFxCopAnalyzers)">
+    <Analyzer Include="$(CodeAnalyzerDir)*.dll" Exclude="$(CodeAnalyzerDir)Microsoft.DotNet.CodeAnalysis.dll"/>
+  </ItemGroup>
+  <PropertyGroup Condition="$(EnableFxCopAnalyzers)">
+    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)Default.ruleset</CodeAnalysisRuleSet>
```
Comment (AlexGhiondea) [inline src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets]: If we add a condition on the CodeAnalysisRuleSet being already set you can have per-project overrides of the ruleset, which might come in handy.

Comment (AlexGhiondea) [inline src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets]: Why not keep the existing ruleset file and add to it the rules we want to have enabled?

Comment (alexperovich) [inline src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets]: I will do that.

Comment (alexperovich) [inline src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets]: It had no rules in it and the name was too specific. I don't care much either way though.

**Pull request merged**